rfcs/taskforce/0002-network-utility.md

```diff
@@ -0,0 +1,203 @@
+# Network Utility Tool
+
+## Goal
+
+When spinning up a network there are a few operations that can and have been automated to decrease iteration speed on testing newer versions of the network.
+
+1. Keyset Generation (Account Creation)
+2. Genesis Ledger Creation
+3. Distribution of custom network configuration and keys
+4. Deployment of infrastructure with updated configuration
+5. Monitoring the health of the network and debuging issues / crashes
+
+This document proposes a tool that unifies these processes by collecting tools already written for these purposes and packaging them in a simple CLI and JS API.
+
+# Design
+
+## Existing System
+
+For these processes we mainly use the [testnet-keys.py](https://github.com/CodaProtocol/coda-automation/blob/master/scripts/testnet-keys.py) script in coda-automation.
+
+1. Keypair generation calls `coda advanced-generate keypair` generate new accounts
+2. Genesis ledger creation happens using custom logic inside this script
+3. Distribution of configuration uses Kubernetes secrets
+4. Deployment consists of a variety of terraform configurations targeting Google Kubernetes Engine
+5. We have a variety of charts and metrics for monitoring macro network health but debugging specific issues is often manual
+
+### Improvements
+
+While this system is effective, there are a variety of potential improvements to this process.
+
+- Keypair generation depends on the daemon
+- Genesis ledger logic could be cleaned up and made more flexible to support a wider variety of configuraitons
+- These tools could be packaged together with a JavaScript API for extending functionality
+- The internal data structures for storing these keypairs, genesis configuration and distribution are not well defined and could be more robust
+- Tools for deployment and monitoring are seperate and these could be packaged in one simple workflow that uses the same underlying functionality
+
+## Proposal
```

Comment:
Just thinking through usage -- I imagine we'll want to write scripts that perform some of the actions described here. There should be some way to take the actions that the interactive commands perform non-interactively. Ideally, through the API rather than the CLI.

If all automation around these actions act through the API and not the CLI then we would just need to write little node wrappers consuming the API if we wanted to, say "load the google form CSV for genesis ledgers, import it as a keyset, generate a genesis ledger with that keyset and a few others, upload everything".

In a way, it's a feature to make the CLI interactive as it forces people not to rely on the human interface for mechanical interactivity.